adapters/repos/db/lsmkv/strategies_map.go

```diff
@@ -324,6 +374,31 @@ func (kv *MapPair) FromBytesReusable(in []byte, keyOnly bool) error {
 	return nil
 }
 
+func (kv *MapPair) FromBytesInverted(in []byte, keyOnly bool) error {
+	var read uint16
+
+	// NOTE: A previous implementation was using copy statements in here to avoid
+	// sharing the memory. The general idea of that is good (protect against the
+	// mmaped memory being removed from a completed compaction), however this is
+	// the wrong place. By the time we are in this method, we can no longer
+	// control the memory safety of the "in" argument. Thus, such a copy must
+	// happen at a much earlier scope when a lock is held that protects against
+	// removing the segment. Such an implementation can now be found in
+	// segment_collection_strategy.go as part of the *segment.getCollection
+	// method. As a result all memory used here can now be considered read-only
+	// and is safe to be used indefinitely.
+
+	kv.Key = in[read : read+8]
```

Comment:
if may be worth validating len(in) >= 8

Reply:
Agreed, I'll add more thorough  validation